src/backend/SecurityContext.c

```diff
@@ -121,17 +122,31 @@ static int BSL_ExecBIBSource(BSL_SecCtx_Execute_f sec_context_fn, BSL_LibCtx_t *
     if (BSL_BundleCtx_ReallocBTSD(bundle, created_block_id, (size_t)encode_result) != BSL_SUCCESS)
     {
         BSL_LOG_ERR("Failed to realloc block ASB space");
+        BSL_AbsSecBlock_Deinit(&abs_sec_block);
         return BSL_ERR_HOST_CALLBACK_FAILED;
     }
 
     if (BSL_BundleCtx_GetBlockMetadata(bundle, created_block_id, &sec_blk) != BSL_SUCCESS)
     {
         BSL_LOG_ERR("Could not get BIB block (id=%lu)", created_block_id);
+        BSL_AbsSecBlock_Deinit(&abs_sec_block);
         return BSL_ERR_SECURITY_OPERATION_FAILED;
     }
 
-    CHK_POSTCONDITION(sec_blk.btsd != NULL);
-    CHK_POSTCONDITION(sec_blk.btsd_len == (size_t)encode_result);
+    // after realloc, original was too small, we must re-encode
+    if (encode_result > BSL_ASB_SMALL_BUFFER_SIZE)
+    {
+        BSL_LOG_INFO("ASB (%d > %d) TOO SMALL, REALLOC NEEDED", encode_result, BSL_ASB_SMALL_BUFFER_SIZE);
+        BSL_Data_InitView(&btsd_view, sec_blk.btsd_len, sec_blk.btsd);
```

Comment:
Does this actually perform any allocation based on `encode_result`? I don't see it.